polymorphia-backend/src/main/java/com/agh/polymorphia_backend/controller/CourseController.java

```diff
@@ -17,10 +19,17 @@
 @RequestMapping("/courses")
 public class CourseController {
     private final CourseService courseService;
+    private final RoadmapService roadmapService;
 
     @GetMapping()
     @PreAuthorize("isAuthenticated()")
     public ResponseEntity<List<AvailableCoursesResponseDto>> getAvailableCourses() {
         return ResponseEntity.ok(courseService.getAvailableCourses());
     }
+
+    @GetMapping("/roadmap")
+    @PreAuthorize("isAuthenticated()")
```

Comment:
```suggestion
    @PreAuthorize("hasAnyAuthority('STUDENT', 'INSTRUCTOR', 'COORDINATOR')")
```

For roadmap, I think this is more specific (`isAuthenticated()` includes `UNDEFINED` role, right?)